build/Build.Steps.cs

```diff
@@ -73,22 +72,41 @@ partial class Build
         {
 
             if (IsWin)
-            {
-                NuGetTasks.NuGetRestore(s => s
-                    .SetTargetPath(Solution)
-                    .SetVerbosity(NuGetVerbosity.Normal)
-                    .When(!string.IsNullOrEmpty(NugetPackageDirectory), o =>
-                        o.SetPackagesDirectory(NugetPackageDirectory)));
-            }
-            else
             {
                 DotNetRestore(s => s
-                    .SetProjectFile(MsBuildProject)
+                    .SetProjectFile(Solution)
                     .SetVerbosity(DotNetVerbosity.Normal)
                     // .SetTargetPlatform(Platform) // necessary to ensure we restore every project
                     .SetProperty("configuration", BuildConfiguration.ToString())
                     .When(!string.IsNullOrEmpty(NugetPackageDirectory), o =>
                         o.SetPackageDirectory(NugetPackageDirectory)));
+
+                var legacyRestoreProjects = Solution.GetNativeProjects()
+                    .Concat(new[] { Solution.GetProject(Projects.Tests.Applications.AspNet) });
+
+                foreach (var project in legacyRestoreProjects)
+                {
+                    // Restore legacy projects
+                    NuGetTasks.NuGetRestore(s => s
+                        .SetTargetPath(project)
+                        .SetSolutionDirectory(Solution.Directory)
+                        .SetVerbosity(NuGetVerbosity.Normal)
+                        .When(!string.IsNullOrEmpty(NugetPackageDirectory), o =>
+                            o.SetPackagesDirectory(NugetPackageDirectory)));
+                }
+            }
+            else
+            {
+                foreach (var project in Solution.GetCrossPlatformManagedProjects())
```

Comment:
What are these projects? At first, it seems redundant with the solution projects.

Reply:
This is the sub collection that was previously done with `.proj`, now the inclusions / exclusions logic is transferred here.
Shortly, all projects that can be restored and built in Linux / MacOS.